### PR TITLE
Finished

### PR DESCRIPTION
**Pull request by radditude**

### CODE REVIEW

File: app/controllers/tweets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    if logged_in
+      erb :'/tweets/new'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if logged_in
+      @user = current_user
```
Comment (Lukeghenco) [inline app/controllers/tweets_controller.rb]: current_user is already saved as an instance variable from the parent class that we are inheriting from, so this just sets two instance variables.

File: app/controllers/tweets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      
+      if params[:content].length <= 140
+        
+        @tweet = Tweet.new(content: params[:content], user_id: @user.id)
```
Comment (Lukeghenco) [inline app/controllers/tweets_controller.rb]: `current_user.tweets.build(params)` is the best way to do this. It creates an associative relationship for us already.

File: app/controllers/tweets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      end
+      
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets' do
+    if logged_in
+      @user = current_user
```
Comment (Lukeghenco) [inline app/controllers/tweets_controller.rb]: see comment on line 13

File: app/controllers/tweets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in
+      @user = current_user
```
Comment (Lukeghenco) [inline app/controllers/tweets_controller.rb]: see comment on line 13

File: app/controllers/tweets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      end
+      
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets/:id' do
+    if logged_in
+      @user = current_user
```
Comment (Lukeghenco) [inline app/controllers/tweets_controller.rb]: see comment on line 13

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class User < ActiveRecord::Base
+  has_many :tweets
+  has_secure_password
+  validates_presence_of :username, :email
```
Comment (Lukeghenco) [inline app/models/user.rb]: you should also check for uniqueness of email and username

**Comment by radditude** [comment]
Whoops, that was a rookie mistake! I fixed the double instance variable problems and the other things you pointed out. Thanks!

**Pull request closed without merging**